types/underscore/underscore-tests.ts

```diff
@@ -209,6 +209,46 @@ _(stooges)
 	.indexBy('age')
 	.value()['40'].age;
 
+let pensioners: string[] = _.chain(stooges)
+    .filter(p => p.age >= 60)
+    .map(p => p.name)
+    .value();
+
+var usersData: _.Dictionary<{ age: number; name: string }> = {
+    'user id': { name: 'moe', age: 40 },
+    'other user Id': { name: 'larry', age: 50 },
+    'fake id': { name: 'curly', age: 60 },
+};
+
+let youngPeopleId: string[] = _.chain(usersData)
+    .map((p, k: string) => k)
+    .value();
+
+let usersTable: { age: number; name: string; id: string }[] = _.chain(usersData)
+    .map<{ age: number; name: string; id: string }>((p, k: string) => {
+        return { id: k, ...p };
+    })
+    .value();
+
+// Test map function with _ChainOfArrays<>
+let usersTable_2 /*: { age: number; name: string; id: string }[][]*/ = _.chain(usersData)
```

Comment:
why did you remove the type here?
It is best to use the flatten() method here and then get the value

Reply:
because the type by the underscore should be like this one in the comment, But with current typescript definition, There is missing one level of array nest. 
`{ age: number; name: string; id: string }[][]  // type return by underscore
{ age: number; name: string; id: string }[]    // type with is calculate base of current definitions
`
It would require more changes in the underscore typescript definition. And I'm not certain what should be changed?
`interface _ChainOfArrays<T> extends _Chain<T[]> `
to 
`interface _ChainOfArrays<T> extends _Chain<T[], T[][]> `
it will not match with data return from function 'groupBy'

Reply:
You are correct. Lets leave it be for now